### PR TITLE
Batch TTL fix + TTL integration tests

### PR DESCRIPTION
**Pull request by angelosantos4**
In this [PR](https://github.com/nodestream-proj/nodestream/pull/272/files) we introduced the option to specify the batch size, whether to execute chunks in parallel along with the retry count for batched APOC queries. This change will allow TTL's to follow a similar pattern of configurability. This PR also introduces an E2E test for ttl's to verify their functionality.

### CODE REVIEW

File: tests/e2e/test_neo4j_pipelines.py [hunk 1]
```diff
@@ -87,3 +122,25 @@ async def test_neo4j_pipeline(
 
         for validator in validations:
             validator(session)
+
+
```
Comment (angelosantos4) [inline tests/e2e/test_neo4j_pipelines.py:126]: Here I only want to create the neo4j container once, run the pipelines that create things, validate them, and then run the TTL pipelines.

File: tests/e2e/test_neo4j_pipelines.py [hunk 2]
```diff
@@ -59,15 +63,186 @@ def validate_fifa_mo_club(session):
...
+        ),
+        {
+            "timestamp": timestamp,
+            "from_node_identifier": from_node_identifier,
+            "to_node_identifier": to_node_identifier,
+        },
+    )
+
+
+def create_test_objects(session: Session):
```
Comment (angelosantos4) [inline tests/e2e/test_neo4j_pipelines.py:180]: The test works like this:
We create 60 nodes in total:
10 ObjectA 60hrs old
10 ObjectB 60hrs old 
10 ObjectA 36hrs old
10 ObjectB 36hrs old 
10 ObjectA 12hrs old
10 ObjectB 12hrs old 

We create 60 pairwise relationships as follows:
ObjectA -ADJACENT_TO {60hrs old} > Object B
ObjectA -CONNECTED_TO {60hrs old} > Object B
....

Following the same pattern. 

With our TTL configuration, were able to first get rid of relationships. 
CONNECTED_TO has 48hr expiry leaving 20 relationships of this type post deletion.
ADJACENT_TO has 24hr expiry leaving 10 relationships of this type post deletion

We assert that the nodes were untouched in the `validate_consistency_in_node_counts ` function located above. 

We then do the exact same thing with the nodes.

Comment (angelosantos4) [reply tests/e2e/test_neo4j_pipelines.py:180]: Oh true true.

File: nodestream_plugin_neo4j/ingest_query_builder.py [hunk 3]
```diff
@@ -254,12 +254,18 @@ def generate_ttl_match_query(self, config: TimeToLiveConfiguration) -> Query:
...
         ttl_match_query = self.generate_ttl_match_query(config)
         operation = (
             DELETE_NODE_QUERY
             if config.graph_object_type == GraphObjectType.NODE
             else DELETE_REL_QUERY
         )
-        return ttl_match_query.feed_batched_query(operation)
+        return ttl_match_query.feed_batched_query(
+            operation, chunk_size, execute_chunks_in_parallel, retries_per_chunk
+        )
```
[the remark this answers is not in the document]
Comment (angelosantos4) [reply nodestream_plugin_neo4j/ingest_query_builder.py:271]: Do we want to add a field into the TTL configuration to override whether to execute in parallel?

**Review by zprobst — APPROVED** [review]

**Pull request merged**